### PR TITLE
feat(wallet) lazy load activity info optimization

### PR DESCRIPTION
**Pull request by stefandunca**
## Updates status-desktop [#11597](https://github.com/status-im/status-desktop/issues/11597)

Trigger async fetching of extra information on each activity filtering request. Only emit the update event for incomplete entries.

Other changes:

- Make DataEntry light as an event payload by making all the fields optional
- Add new required fields to the activity DataEntry

### CODE REVIEW

**Comment by stefandunca** [comment]
Given that the hanging issue is only reproducible for a specific test account in test mode, I created a follow-up issue to address it: https://github.com/status-im/status-desktop/issues/12064

File: services/wallet/activity/activity.go [hunk 1]
```diff
@@ -174,6 +195,42 @@ func (e *Entry) PayloadType() PayloadType {
...
+			},
+			TokenID: tokenIDToWalletBigInt(e.tokenIn.TokenID),
+		}
+	} else if e.tokenOut != nil {
+		return &thirdparty.CollectibleUniqueID{
+			ContractID: thirdparty.ContractID{
+				ChainID: e.tokenOut.ChainID,
+				Address: e.tokenOut.Address,
+			},
+			TokenID: tokenIDToWalletBigInt(e.tokenIn.TokenID),
```
Comment (dlipicar) [inline services/wallet/activity/activity.go]: shouldn't this be `tokenOut`?

**Review by dlipicar — APPROVED** [review]
Other than my comment, looks good!

File: services/wallet/activity/activity.go (hunk 1 above)
Comment (stefandunca) [reply services/wallet/activity/activity.go]: 🤦🏻 Thanks, @dlipicar for the eagle eyes 👀

**Pull request merged**